### DISCUSSION TITLE
位运算的请教继续

### DISCUSSION THREAD

**Discussion by KleinSteven** [Codesire-Deng/co_context#81] [Q&A] [answered]
抱歉，由于我这边的问题，无法在源discussion回复，所以新开一个讨论
io_info.as_user_data()的低位不会造成影响吗，只有在其低位为0是才成立不是吗？比如我通过debug得到的值为93824992375544, 它的低3位确实是0

**Comment by Codesire-Deng** [comment, marked as answer]
`io_info.as_user_data()` 的低 3 位保证为 0，[这是由 static_assert 确保的](https://github.com/Codesire-Deng/co_context/blob/6edd7021ec7e36e2a40b8cc17759f880d8676108/include/co_context/detail/task_info.hpp#L22-L25)，进而后面的位运算可以成立。

原理是在 64 位机器上，io_info 内含一个指针，所以至少是按 8 字节对齐的，其地址的低 3 位总是 0。

**Reply by KleinSteven** [reply]
好的，感谢解答
